### PR TITLE
boj 1644 소수의 연속합 :sos:

### PR DESCRIPTION
**Pull request by saechimdaeki**
### :star: 문제 링크 :book:
- https://www.acmicpc.net/problem/1644
---

### :star2: 사용한 알고리즘
- 에라토스테네스의 체 + 투포인터 
---

### :stars: 코드
```
import java.io.*;
import java.util.*;
import java.util.stream.IntStream;

public class Main {
    public static void main(String[] args) throws IOException {
        BufferedReader br=new BufferedReader(new InputStreamReader(System.in));
        int n=Integer.parseInt(br.readLine());
        ArrayList<Integer> list=new ArrayList<>();
        boolean[] visited=new boolean[n+1];
        IntStream.rangeClosed(2, n).filter(i -> !visited[i]).forEach(i -> {
            list.add(i);
            IntStream.iterate(i * 2, j -> j <= n, j -> j + i).forEachOrdered(j -> visited[j] = true);
        });
        int left=0,right=0,answer=0;
        int sum=0;
        if(list.size()!=0)
            sum=list.get(0);
        while (left<=right &&right<list.size()){
            if(sum<=n){
                if(sum==n)
                    answer++;
                right++;
                if(right<list.size())
                    sum+= list.get(right);
            }else {
                sum -= list.get(left);
                left++;
            }
        }
        System.out.print(answer);
    }
}
```
---

### CODE REVIEW

**Comment by saechimdaeki** [comment]
처음에는 이러한 stream방식이 아니였는데 java8에서는 제공하지 않아 백준에서는 컴파일에러가 뜨더라고용... 어찌어찌하다보니 stream api가 이렇게 되었습니다..

File: saechimdaeki/boj1644소수의연속합.java [hunk 1]
```diff
@@ -0,0 +1,33 @@
+import java.io.*;
+import java.util.*;
+import java.util.stream.IntStream;
+
+public class boj1644소수의연속합 {
```
Comment (aimiebell) [inline saechimdaeki/boj1644소수의연속합.java:5]: I detect that this code is problematic. According to the [Bad practice (BAD_PRACTICE)](https://spotbugs.readthedocs.io/en/stable/bugDescriptions.html#bad-practice-bad-practice), [Nm: Class names should start with an upper case letter (NM_CLASS_NAMING_CONVENTION)](https://spotbugs.readthedocs.io/en/stable/bugDescriptions.html#nm-class-names-should-start-with-an-upper-case-letter-nm-class-naming-convention).
Class names should be nouns, in mixed case with the first letter of each internal word capitalized. Try to keep your class names simple and descriptive. Use whole words-avoid acronyms and abbreviations (unless the abbreviation is much more widely used than the long form, such as URL or HTML).

Comment (saechimdaeki) [reply saechimdaeki/boj1644소수의연속합.java:5]: This code is an algorithm code from a site called Baek Jun.
The java class on the site must be a class named Main unconditionally, and it is kept when submitted.

However, this repository is simply for viewing the algorithm-solving logic and does not care about such matters to classify the problem.

Comment (saechimdaeki) [reply saechimdaeki/boj1644소수의연속합.java:5]: and i updated comment code not code file

Comment (saechimdaeki) [reply saechimdaeki/boj1644소수의연속합.java:5]: I simply didn't care about the class name, but thanks for the feedback, I will definitely do that next time!

**Pull request merged**